L1Trigger/DTTriggerPhase2/src/MPQualityEnhancerFilterBayes.cc

```diff
@@ -0,0 +1,261 @@
+#include "L1Trigger/DTTriggerPhase2/interface/MPQualityEnhancerFilterBayes.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+using namespace edm;
+using namespace std;
+using namespace cmsdt;
+;
+
+// ============================================================================
+// Constructors and destructor
+// ============================================================================
+MPQualityEnhancerFilterBayes::MPQualityEnhancerFilterBayes(const ParameterSet &pset)
+    : MPFilter(pset), debug_(pset.getUntrackedParameter<bool>("debug")) {}
+
+// ============================================================================
+// Main methods (initialise, run, finish)
+// ============================================================================
+void MPQualityEnhancerFilterBayes::initialise(const edm::EventSetup &iEventSetup) {}
+
+void MPQualityEnhancerFilterBayes::run(edm::Event &iEvent,
+                                       const edm::EventSetup &iEventSetup,
+                                       std::vector<metaPrimitive> &inMPaths,
+                                       std::vector<metaPrimitive> &outMPaths) {
+  filterCousins(inMPaths, outMPaths);
+  if (debug_) {
+    LogDebug("MPQualityEnhancerFilterBayes") << "Ended Cousins Filter. The final primitives before Refiltering are: ";
+    for (unsigned int i = 0; i < outMPaths.size(); i++) {
+      printmP(outMPaths[i]);
+    }
+    LogDebug("MPQualityEnhancerFilterBayes") << "Total Primitives = " << outMPaths.size();
+  }
+}
+
+void MPQualityEnhancerFilterBayes::finish(){};
+
+///////////////////////////
+///  OTHER METHODS
+int MPQualityEnhancerFilterBayes::areCousins(metaPrimitive mp, metaPrimitive second_mp) {
+  DTSuperLayerId mpId(mp.rawId);
+  DTSuperLayerId second_mpId(second_mp.rawId);
+
+  int output = 0;
+  if (mpId.wheel() != second_mpId.wheel() || mpId.station() != second_mpId.station() ||
+      mpId.sector() != second_mpId.sector()) {
+    return output;
+  }
+
+  if (mp.wi1 == second_mp.wi1 and mp.tdc1 == second_mp.tdc1 and mp.wi1 != -1 and mp.tdc1 != -1)
```

Comment:
Just asking you to check that this is the correct logic you have in mind: are all these options exclusive (and in that case better to have `else if`'s instead) or the 8th one will overwrite all previous ones (in which case the logic is correct, but it could be better to write them in reverse order with some `else if` statement)?